src/opcode.rs

```diff
@@ -641,6 +641,36 @@ opcode! {
     }
 }
 
+opcode! {
+    /// Set a socket option.
+    pub struct SetSockOpt {
+        fd: { impl sealed::UseFixed },
+        level: { u32 },
+        optname: { u32 },
+        optval: { *const libc::c_void },
+        optlen: { u32 },
+        ;;
+        flags: u32 = 0
+    }
+
+    pub const CODE = sys::IORING_OP_URING_CMD;
+
+    pub fn build(self) -> Entry {
+        let SetSockOpt { fd, level, optname, optval, optlen, flags } = self;
+        let mut sqe = sqe_zeroed();
+        sqe.opcode = Self::CODE;
+        assign_fd!(sqe.fd = fd);
+        sqe.__bindgen_anon_1.__bindgen_anon_1.cmd_op = sys::SOCKET_URING_OP_SETSOCKOPT;
```

Comment:
Should we expose `cmd_op`? Provide generalized `CmdSock` opcode instead of specific `SetSockOpt`.

`SOCKET_URING_OP_SETSOCKOPT` can be exposed as a `SocketOp` in `types.rs`.
like

```rust
pub struct SocketOp(u32);

impl SocketOp {
    pub const SETSOCKOPT: SocketOp = SocketOp(sys::SOCKET_URING_OP_SETSOCKOPT);
}
```

Reply:
Not sure what you mean. `level`, `optname`, `optval`, and `optlen` are all `setsockopt` specific, so we'd want an opcode creator either way.

Non-`setsockopt` needs other parameters.

Sure, it'd be possible to add an abstraction layer inside this crate to deduplicate, but honestly the duplicate code is like one single simple assignment line.

Maybe I'm not understanding your suggestion?

Reply:
Ok, in any case there are no other needs at moment. new opcode can be added in future if needed.